core/src/database/query/QueryBuilder.cpp

```diff
@@ -133,5 +137,10 @@ namespace ledger {
             return *this;
         }
 
+        QueryBuilder &QueryBuilder::groupBy(std::string &&group) {
```

Comment:
Passing an argument as rvalue reference is rarely necessary and often considered as over-engineering. Here `QueryBuilder` is telling "Hello, I need to own the full string resource to perform the groupBy" where it should tell "Hello, I'm going to use my own resource based on what you are giving to me". In that sense, it should expect a value (`QueryBuilder &QueryBuilder::groupBy(std::string group)`). This still allows to pass an rvalue reference (by moving the given string, or passing an rvalue) and allow type erasure. You can then move your local copy to your resource as you did.

Reply:
Right, I just copied&renamed an other method without more thinking...